tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter.cc

```diff
@@ -633,6 +639,96 @@ StatusOr<bool> FuseRelu(HloComputation* comp) {
   return changed;
 }
 
+StatusOr<bool> FuseRelu6(HloComputation* comp, se::CudaComputeCapability cc) {
+  bool changed = false;
+  for (HloInstruction* instr : comp->MakeInstructionPostOrder()) {
+    if (!IsCudaComputeCompatible(instr, cc)) {
+      return false;
+    }
+    HloInstruction *gte, *conv;
+    // We don't want to upgrade depthwise convolutions to ConvBiasActivation,
+    // because the fused CUDNN functions are slower for some of those.
+    if (!Match(instr,
+               m::Clamp(
+                   m::Broadcast(m::ConstantEffectiveScalar(0)),
+                   m::GetTupleElement(
+                       &gte, m::Op(&conv)
+                                 .WithPredicate(IsNonDepthwiseConvCustomCall)
+                                 .WithOneUse())
+                       .WithElementType(F16)
```

Comment:
Can you explain why this is enabled only for fp16 convs?  Is the intent for this to exclude fp32 (why?) and include fp16/bf16/int8?  Or do we really only want fp16?

Reply:
Sure. Currently we only want to support fp16. For other dtypes, we might need to have more code changes: e.g., (1) for fp32, we need to support NHWC data layout in XLA for Ampere+ GPUs and check TF32 is enabled. I think currently we use NCHW by default. (2) bf16 should be easily supported as fp16, but I remembered the cudnn has a better support of bf16 only after a certain version (need to double check). (3) Not very familiar with the int8 support, but since it is supported according to the doc, I think we could do it as well.

Reply:
Thanks, @kaixih.

Should we change layout assignment if Ampere+ tf32 wants NHWC?

I guess now that I think about it, nobody should be doing relu6 on an int8 conv, that is very silly.  :)  So I guess tf32, f32, and bf16 are the only datatypes to worry about.

Reply:
> Should we change layout assignment if Ampere+ tf32 wants NHWC?
Yes, I think this is correct.

Reply:
Thanks, I've filed an internal bug on this (Google b/290983064 for those following along).